pkg/registry/flowcontrol/rest/storage_flowcontrol.go

```diff
@@ -174,106 +181,89 @@ func shouldEnsureSuggested(flowcontrolClientSet flowcontrolclient.FlowcontrolV1b
-		}
-		if err != nil {
-			return fmt.Errorf("cannot create mandatory FlowSchema %s due to %v", expectedFlowSchema.Name, err)
-		}
-		klog.V(3).Infof("Created mandatory FlowSchema %s", expectedFlowSchema.Name)
+func removeConfiguration(clientset flowcontrolclient.FlowcontrolV1beta1Interface) error {
+	if err := removeFlowSchema(clientset.FlowSchemas()); err != nil {
+		return err
 	}
-	for _, expectedPriorityLevelConfiguration := range priorityLevels {
-		actualPriorityLevelConfiguration, err := flowcontrolClientSet.PriorityLevelConfigurations().Get(context.TODO(), expectedPriorityLevelConfiguration.Name, metav1.GetOptions{})
-		if err == nil {
-			// TODO(yue9944882): extract existing version from label and compare
-			// TODO(yue9944882): create w/ version string attached
-			wrongSpec, err := priorityLevelHasWrongSpec(expectedPriorityLevelConfiguration, actualPriorityLevelConfiguration)
-			if err != nil {
-				return fmt.Errorf("failed checking if mandatory PriorityLevelConfiguration %s is up-to-date due to %v, will retry later", expectedPriorityLevelConfiguration.Name, err)
-			}
-			if wrongSpec {
-				if _, err := flowcontrolClientSet.PriorityLevelConfigurations().Update(context.TODO(), expectedPriorityLevelConfiguration, metav1.UpdateOptions{FieldManager: thisFieldManager}); err != nil {
-					return fmt.Errorf("failed upgrading mandatory PriorityLevelConfiguration %s due to %v, will retry later", expectedPriorityLevelConfiguration.Name, err)
-				}
-				klog.V(3).Infof("Updated mandatory PriorityLevelConfiguration %s because its spec was %#+v but must be %#+v", expectedPriorityLevelConfiguration.Name, actualPriorityLevelConfiguration.Spec, expectedPriorityLevelConfiguration.Spec)
-			}
-			continue
-		}
-		if !apierrors.IsNotFound(err) {
-			return fmt.Errorf("failed getting PriorityLevelConfiguration %s due to %v, will retry later", expectedPriorityLevelConfiguration.Name, err)
-		}
-		_, err = flowcontrolClientSet.PriorityLevelConfigurations().Create(context.TODO(), expectedPriorityLevelConfiguration, metav1.CreateOptions{FieldManager: thisFieldManager})
-		if apierrors.IsAlreadyExists(err) {
-			klog.V(3).Infof("Mandatory PriorityLevelConfiguration %s already exists, skipping creating", expectedPriorityLevelConfiguration.Name)
-			continue
-		}
-		if err != nil {
-			return fmt.Errorf("cannot create mandatory PriorityLevelConfiguration %s due to %v", expectedPriorityLevelConfiguration.Name, err)
-		}
-		klog.V(3).Infof("Created mandatory PriorityLevelConfiguration %s", expectedPriorityLevelConfiguration.Name)
+
+	return removePriorityLevel(clientset.PriorityLevelConfigurations())
+}
+
+func removeFlowSchema(client flowcontrolclient.FlowSchemaInterface) error {
+	bootstrap := append(flowcontrolbootstrap.MandatoryFlowSchemas, flowcontrolbootstrap.SuggestedFlowSchemas...)
+	candidates, err := ensurer.GetFlowSchemaRemoveCandidate(client, bootstrap)
+	if err != nil {
+		return err
 	}
-	return nil
+	if len(candidates) == 0 {
+		return err
```

Comment:
return nil

Reply:
[technically it doesn't matter - but it's misleading to the user]

Reply:
oops, it's a copy paste error, fixed.